src/halmos/sevm.py

```diff
@@ -1654,7 +1655,14 @@ def balance_of(self, addr: Word) -> Word:
         self.path.append(Select(EMPTY_BALANCE, addr) == ZERO)
 
         # practical assumption on the max balance per account
-        self.path.append(ULT(value, con(2**96)))
+        # (only for symbolic values, otherwise we may append false constraints)
+        if is_bv_value(value):
+            if (v := value.as_long()) > MAX_ETH:
+                warn(
+                    f"address {render_address(addr)} has balance {v} greater than {MAX_ETH=}"
+                )
+        else:
+            self.path.append(ULE(value, con(MAX_ETH)))
```

Comment:
just note:  if we decide to raise in the true branch, it would be more consistent to check the path condition here and raise as well if it's infeasible. but it might incur too much overhead to call the smt solver everytime.

Reply:
I see, `ULE(value, con(MAX_ETH))` may get simplified to `False` and then we'd still get the error message... 🫠

Reply:
can you check again with https://github.com/a16z/halmos/pull/501/commits/a6229542db6483a4de4d7f5da7133b457738c110 ?

Reply:
looks good to me!